### DISCUSSION TITLE
How to build HTML files from a storybook project?

### DISCUSSION THREAD

**Discussion by Wipster** [storybookjs/storybook#29009] [Help]
### Summary

I created a storybook project via npx storybook@latest init and chose html as the type and html-vite as the bundler. I want to use storybook to distribute the templates to different production apps like outlined in this article: [https://storybook.js.org/tutorials/design-systems-for-developers/react/en/distribute/](https://storybook.js.org/tutorials/design-systems-for-developers/react/en/distribute/)

Now after the initial storybook setup was done I discovered the actual templates to be js files. I understand that this goes well with the technology because the stories themselves are js files but in the end I want to distribute plain HTML templates to projects that don't use any js frontend framework.

I didn't find any, but is there a command I can use? Otherwise what does html-vite actually do at all? What additional steps do I need to take in order to build html templates from my stories?

**Comment by Marklb** [comment]
I only skimmed the tutorial, so I could have missed something, but I think it is talking about publishing a storybook app for composing in another app's storybook app. It sounds like you are trying to use stories as part of a non-storybook app, if I understand correctly.

Storybook isn't a build tool for applications to consume components from. You could embed or render the stories in an app, but it would still need to use js to render the story. Even if you don't use Storybook's renderer, the story itself is sort of like a factory function that knows how to render the component, but there may be some functions that need to be run before that story instance finishes rendering.

When viewing a story docs, there is a "Show Code" button that can show an HTML snippet, but it is from the perspective of a documentation app. So, it is mainly an example of the code to render the same thing as the story. That code is ideally what you could copy and paste in an application to render the same thing as the story.
